pkg/asset/installconfig/azure/validation.go

```diff
@@ -89,23 +94,35 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 
 	// Default requirements need to be sufficient to support Control Plane instances.
 	defaultInstanceReq := controlPlaneReq
+	defaultDiskType := ""
```

Comment:
```suggestion
	defaultDiskType := "Premium_LRS"
```
As ugly as it is, the default is actually defined with the machines assets.
https://github.com/openshift/installer/blob/aa9ff594a8d582acf687303f18345ccc39b106b4/pkg/asset/machines/worker.go#L97

Let's define the default in a constant in `pkg/types/azure/machinepool.go`.